tests/integration/checkpoint.bats

```diff
@@ -441,3 +441,38 @@ function simple_cr() {
 	pid=$(cat "pid")
 	grep -q "${REL_CGROUPS_PATH}$" "/proc/$pid/cgroup"
 }
+
+@test "checkpoint/restore and exec" {
+	runc run -d --console-socket "$CONSOLE_SOCKET" test_busybox
+	[ "$status" -eq 0 ]
+
+	testcontainer test_busybox running
+
+	local execed_pid=""
+	for _ in $(seq 2); do
+		# checkpoint the running container
+		runc checkpoint --work-path ./work-dir test_busybox
+		[ "$status" -eq 0 ]
+
+		# after checkpoint busybox is no longer running
+		testcontainer test_busybox checkpointed
+
+		# restore from checkpoint
+		runc restore -d --work-path ./work-dir --console-socket "$CONSOLE_SOCKET" test_busybox
+		[ "$status" -eq 0 ]
+
+		# busybox should be back up and running
+		testcontainer test_busybox running
+
+		# verify that previously exec'd process is restored.
+		if [ -n "$execed_pid" ]; then
+			runc exec test_busybox ls -ld "/proc/$execed_pid"
+			[ "$status" -eq 0 ]
+		fi
+
+		# exec a new background process.
+		runc exec test_busybox sh -c 'sleep 1000 < /dev/null &> /dev/null & echo $!'
```

Comment:
Why do we have the dev/null redirections here? I guess there is a criu thing that is better with that? Can you please elaborate?

Reply:
Same as in `simple_cr` from where the code is copied. The issue here is we need `sleep` process to be checkpointable, and if some of its file descriptors point to files/pipes on the host, it is not checkpointable (as the process is not contained in a container).

Reply:
Hmm, I don't see this null redirections in checkpoints.bat. Also, runc does a reopen of `/dev/null`, so I guess that shouldn't be the issue?

But yeah, removing those redirections locally makes the test get stalled. I guess this is fine if criu devs think it is needed. Although some explanation on why that is the case would be great, IMHO.

Reply:
This sleep process is daemonized, and `runsc exec` exits right after forking it. If we don't redirect its file descriptors, they will be closed when runsc exec exits, and the sleep process can be killed by SIGHUP.